### DISCUSSION TITLE
'Infeasible problem (MOSEK)' for LQR based on YALMIP solution

### DISCUSSION THREAD

**Discussion by elkhatibkamal** [yalmip/YALMIP#1020] [Problems using YALMIP] [answered]
Hi Johan,
I have a bunch of LMI constraints and I wrote codes as follows and solved them by   LQR with (K = lqr(Aaug,Baug,Q,R)) we obtain results.

But when we use the same Q and R weight matrices and solve LQR problem by the mosek toolbox. The results show that the problem is 'Infeasible problem (MOSEK)',
Thank you in advance for your help.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

clear all
close all
format short
clc

   A =[-0.0333    0.0064    0.0064    0.0064    0.0018
    0.0064   -0.0333    0.0064    0.0064    0.0018
    0.0064    0.0064   -0.0333    0.0064    0.0018
    0.0064    0.0064    0.0064   -0.0333    0.0020
    0.0497    0.0497    0.0497    0.0497   -0.9866];

    C =[8.6432   -2.7497   -2.7497   -2.7435   -1.7021
   -2.7497    8.6432   -2.7497   -2.7435   -1.7021
   -2.7497   -2.7497    8.6432   -2.7435   -1.7021
   -2.7497   -2.7497   -2.7497    8.6494   -1.7021
    0.1700    0.1700    0.1700    0.1700    0.0850];

   B = [0.0035         0         0         0         0
         0    0.0035         0         0         0
         0         0    0.0035         0         0
         0         0         0    0.0035         0
         0         0         0         0    0.1005];
D = [0];
m=5; 
Aaug=[    A         zeros(m,m) ; 
          -C         zeros(m,m)]; 
   
eigv_AA=eig(Aaug)
 
Baug=[    B        ;
    zeros(m,m) ];
Caug=[zeros(m,m)             eye(m)];

Daug=0;

R=[50 0 0 0 0
   0 50 0 0 0
   0 0 50 0 0
   0 0 0 50 0
   0 0 0 0 800]; 
 Q =[   6996     0      0      0      0      0      0     0     0          0;
         0    6996     0      0      0      0      0     0     0          0;
         0      0     6996     0      0      0      0     0     0          0;
         0      0      0    6996     0      0      0     0     0          0;
         0      0      0      0     0.1    0      0     0     0          0;
         0      0      0      0      0     0.010     0     0     0          0;
         0      0      0      0      0      0     0.010    0     0          0;
         0      0      0      0      0      0      0    0.010    0          0;
         0      0      0      0      0      0      0     0     0.010       0;
         0      0      0      0      0      0      0     0     0       1.64];

%%%%%%%%%%%%%  solving Based on LQR

K = lqr(Aaug,Baug,Q,R)
%%%%%%%%%%%%%%%%% solving Based on mosek toolbox
n1 = size(Aaug,1);
n2= size(Baug,1);
P = sdpvar(n1,n1);
L1 =P >0;
L4=[[Aaug'*P+P*Aaug+Q  P*Baug;Baug'*P -R]<0];
L=L1+L4;
opts1 = sdpsettings('solver','mosek');
sol1= optimize(L,[],opts1);
P= value(P);
K =inv(R)*Baug'*P;
sol1.info

**Comment by johanlofberg** [comment, marked as answer]
Smells like you are doing the same error as here
https://math.stackexchange.com/questions/2265647/algebraic-riccati-inequality-solution-via-lmi

**Reply by elkhatibkamal** [reply]
Dear Prof Johan

Many thanks for your reply, but the answer is not clear for me
my question is,
 when I solved by LQR with (K = lqr(Aaug,Baug,Q,R)) we obtain results.
and  when we use the same Q and R weight matrices and solve LQR problem by the mosek toolbox. The results show that the problem is 'Infeasible problem (MOSEK)',

**Reply by johanlofberg** [reply]
Did you read the link? You have either precisely as in the link applied a Schur complement on the wrong ARE, or you have applied an incorrect Schur complement on the correct ARE. LQR cannot be cast as an LMI in P, it requires working in both P and K, and variable changes.
